base/linalg/uniformscaling.jl

```diff
@@ -53,31 +53,39 @@ for (t1, t2) in ((:UnitUpperTriangular, :UpperTriangular),
             end
         end
     end
+end
 
-    function (-)(J::UniformScaling, UL::Union{UpperTriangular,UnitUpperTriangular})
-        ULnew = similar(UL, promote_type(eltype(J), eltype(UL)))
-        n = size(ULnew, 1)
-        ULold = UL.data
-        for j = 1:n
-            for i = 1:j - 1
-                ULnew[i,j] = -ULold[i,j]
-            end
-            ULnew[j,j] = J.λ - ifelse(isa(UL, UnitUpperTriangular), 1, ULold[j,j])
+function (-)(J::UniformScaling, UL::Union{UpperTriangular,UnitUpperTriangular})
+    ULnew = similar(UpperTriangular(full(UL)), promote_type(eltype(J), eltype(UL)))
```

Comment:
Just leave out the `UpperTriangular` call here. Only the `Matrix` is needed and indexing into a `Matrix` is probably faster than into a `UpperTriagular`.